File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,char>mp;
        unordered_map<char,char>vis;
        for(int i = 0 ; i < s.size();i++){
            if(mp.find(s[i])==mp.end()){
                if(vis[t[i]])return false;
                mp[s[i]]=t[i];
            }
            else if(mp[s[i]] != t[i] ) return false;
            vis[t[i]]=1;
            s[i]=mp[s[i]];
        }
        cout<<s<<" "<<t<<endl;
        return s==t;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        dic = {}
        vis = {}
        s=list(s)
        t=list(t)
        for i in range(0,len(s)):
            if (s[i] not in dic) :
                if(t[i] in vis) :
                    return False
                dic[s[i]] = t[i]
                vis[t[i]] = 1
            s[i] = dic[s[i]]

              
        
        
        return s == t;